src/rdb/rdb_tx.c

```diff
@@ -211,18 +211,17 @@ rdb_tx_hdr_decode(const void *buf, size_t len, struct rdb_tx_hdr *hdr)
 static bool
 rdb_tx_is_critical(struct rdb_tx *tx)
 {
-	struct rdb_tx_hdr	hdr;
-	bool			crit = true;
+	struct rdb_tx_hdr hdr = {.critical = 1};
 
 	D_ASSERT(tx != NULL);
-	if (tx->dt_entry) {
-		ssize_t		nb;
+	if (tx->dt_entry_len > 0) {
+		ssize_t nb;
 
+		D_ASSERT(tx->dt_entry != NULL);
 		nb = rdb_tx_hdr_decode(tx->dt_entry, tx->dt_entry_len, &hdr);
-		D_ASSERT(nb == sizeof(struct rdb_tx_hdr));
-		crit = hdr.critical;
+		D_ASSERTF(nb > 0, "nb=%zd\n", nb);
```

Comment:
A decoder here, such as `rdb_tx_hdr_decode`, takes a sequence of bytes, and outputs how many bytes have been consumed as well as the decoded data. The number of consumed bytes may or may not be equal to the size of the decoded data, although they are indeed equal in this case at the moment. For instance, if we were crazy enough to encode in JSON... Hence, I think it's better to avoid using the internal knowledge (i.e., the number of bytes consumed) outside the decoder in non-test code.